cre.cpp

```diff
@@ -3495,6 +3495,29 @@ static int getImageDataFromPosition(lua_State *L) {
     return 0;
 }
 
+static int setUserHyphenationDict(lua_State *L) {
+    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
+    const char *filename = luaL_checkstring(L, 2);
+    UserHyphenDict::init(filename);
+    return 0;
+}
+
+static int getHyphenation(lua_State *L) {
+    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
+    const char *word = luaL_checkstring(L, 2);
+    lString32 hyphenation = UserHyphenDict::getHyphenation(word);
+    lua_pushstring(L, UnicodeToLocal(hyphenation).c_str());
+    return 1;
+}
+
+static int formatHyphenation(lua_State *L) {
```

Comment:
Can we have clearer names ?
getUserHyphenationForWord ? getUserHyphenationFormForWord ?
getFormattedUserHyphenationForWord ?

Not sure what are the differences between all these :) Is formatHyphenation() to just not have to add utf8proc ffi stuff ? (which is fine as long as the function name hides this fact :)